config/tgl-h-cavs.toml

```diff
@@ -240,7 +240,7 @@ count = 15
 	uuid = "e61bb28d-149a-4c1f-b709-46823ef5f5ae"
 	affinity_mask = "0xF"
 	instance_count = "10"
-	domain_types = "1"
+	domain_types = "0"
```

Comment:
I fail to see how an SRC qualifiers for a LL component? It's a heavy duty processing that is used to match rates that are precisely not aligned with the LL processing, so is this not going to introduce variability in the LL domain?

We want to keep the low-latency domain with as few heavy-duty processing as possible, precisely meet strict latency requirements.

Nothing prevents the SRC from being executed in a background task with lower priority and larger buffers.

Lost on this one.

Reply:
@plbossart We have no DP support on tgl in SOF mainline, so this either this is put to LL or disabled.

Reply:
Right, but let's agree that this is not the long-term direction. When DP is enabled, we have to revert this change.

Reply:
@plbossart src is not a typical DP.
yes, it is a heavy processing, but with same period as LL
Typical DPs are - voice recognition (>300ms processing period), MPEG decoders (26ms for mp3), etc. 
As long as it is on the same core that rest of the pipeline, there's no point at all to use DP for it
DP processing is much heavier in fact (a thread!)

DP may be enabled on TGL, but it has never been tested there

Reply:
@plbossart We could already put SRC to DP in mtl.toml. Please @singalsu chime in but I was under impression SRC is LL. It can handle audio in LL-sized chunks, has fixed processing time (relation to amount of input data), so I'd imagine it can meet LL needs.

Reply:
> @plbossart src is not a typical DP. yes, it is a heavy processing, but with same period as LL Typical DPs are - voice recognition (>300ms processing period), MPEG decoders (26ms for mp3), etc. As long as it is on the same core that rest of the pipeline, there's no point at all to use DP for it DP processing is much heavier in fact (a thread!)
 
If we have MP3 decoder + SRC, this entire chain does not need to be low-latency. It'll have extremely variable processing requirements depending on the source encoding.

Likewise if we have a 'media' pipeline with PCM deep-buffer + SRC, what is the point of the SRC being in the LL domain? At this point you've on-purpose increased the buffering to reduce power, latency was never a concern.

I don't see what benefits we get here by using LL... in my experience low-latency can only be done by moving all the cruft in the background - in other words do few things and do them well.

Reply:
Right, but the 'media' usages are not really low-latency, as measured by the latency added by the decoding process or the deep-buffer.

The problem is that we have one module used in completely different use cases. So if we do MP3 decode -> SRC -> 3rd party processing, the SRC would be tagged as LL even though the source and sink are not.

The point is that the arbitrary decision on the module domain does not make sense here. If you have a processing chain, going back and forth between LL and DP is questionable at best.

Reply:
@plbossart Your review is blocking merge of this (I re-asked review from you yesterday that cleared the red x). Can you reconsider?